src/Components/MapComponent/MapComponent.jsx

```diff
@@ -147,9 +162,116 @@ class MapComponent extends Component {
       onSetCurrentStopsGeoJSON(newCurentStopsGeoJSON);
     });
 
+    const modify = new Modify({
+      source: this.routeVectorSource,
+      pixelTolerance: 2,
+      style: () => {
+        const { currentMot } = this.props;
+        return pointStyleFunction(currentMot);
+      },
+    });
+
+    modify.on('modifystart', evt => {
+      // save start point to find where to add the new HOP!
+      this.initialRouteDrag = {
+        features: evt.features.getArray(),
+        coordinate: evt.mapBrowserEvent.coordinate,
+      };
+    });
+
+    modify.on('modifyend', evt => {
+      const { features } = this.initialRouteDrag;
+      const {
+        currentMot,
+        currentStops,
+        currentStopsGeoJSON,
+        onSetCurrentStops,
+        onSetCurrentStopsGeoJSON,
+      } = this.props;
+      const updatedCurrentStops = _.clone(currentStops);
```

Comment:
where does `_clone` comes from ?

Reply:
To clone a deep ol objects and be able to change its params and dispatch it.
It is already used widely in the repo so I use it as well, to avoid state update pb.

Reply:
ok